wremnants/CardTool.py

```diff
@@ -664,67 +675,68 @@ def setOutput(self, outfolder, fitvars=[], doStatOnly=False, postfix=None):
             prefix = "ZMassDilepton"
         if self.datagroups.lowPU:
             prefix += "_lowPU"
-
         tag = prefix+"_"+"_".join(fitvars)
         if doStatOnly:
             tag += "_statOnly"
         if self.datagroups.flavor:
             tag += f"_{self.datagroups.flavor}"
         if postfix is not None:
             tag += f"_{postfix}"
-
         self.outfolder = f"{outfolder}/{tag}/"
         if not os.path.isdir(self.outfolder):
             os.makedirs(self.outfolder)
-
         suffix = f"_{self.datagroups.flavor}" if self.datagroups.flavor else ""
         if self.xnorm:
             suffix += '_xnorm'
 
         self.cardName = (f"{self.outfolder}/{prefix}_{{chan}}{suffix}.txt")
         self.setOutfile(os.path.abspath(f"{self.outfolder}/{prefix}CombineInput{suffix}.root"))
-            
-    def writeOutput(self, args=None, xnorm=False, forceNonzero=True, check_systs=True, simultaneousABCD=False):
-        self.xnorm = xnorm
+
+    def writeOutput(self, args=None, forceNonzero=True, check_systs=True):
         self.datagroups.loadHistsForDatagroups(
             baseName=self.nominalName, syst=self.nominalName,
             procsToRead=self.datagroups.groups.keys(),
             label=self.nominalName, 
             scaleToNewLumi=self.lumiScale, 
             forceNonzero=forceNonzero,
-            fakerateIntegrationAxes=self.fakerateIntegrationAxes)
-        if simultaneousABCD and not xnorm:
+            sumFakesPartial=not self.ABCD,
+            fakerateIntegrationAxes=self.getFakerateIntegrationAxes())
+        if self.ABCD and not self.xnorm:
             setSimultaneousABCD(self)
+        
         self.writeForProcesses(self.nominalName, processes=self.datagroups.groups.keys(), label=self.nominalName, check_systs=check_systs)
         self.loadNominalCard()
-        if self.pseudoData and not xnorm:
-            self.addPseudodata([x for x in self.datagroups.groups.keys() if x != "Data"],
-                               [x for x in self.datagroups.groups.keys() if x != "Data" and not self.pseudoDataProcsRegexp.match(x)])
+        if self.pseudoData and not self.xnorm:
+            self.addPseudodata([x for x in self.datagroups.groups.keys() if x != self.getDataName()],
+                               [x for x in self.datagroups.groups.keys() if x != self.getDataName() and not self.pseudoDataProcsRegexp.match(x)])
 
         self.writeLnNSystematics()
         for syst in self.systematics.keys():
             if self.isExcludedNuisance(syst): continue
             systMap = self.systematics[syst]
             systName = syst if not systMap["name"] else systMap["name"]
             processes = systMap["processes"]
+            if systMap["fromNominal"]:
```

Comment:
I think this name could be more description, effectively this is controlling whether you vary the syst through the fakes, right?

Reply:
No, it's to create the systematic variation starting from the nominal histogram, e.g. by applying the "actionMap". 
Do you have a suggestion for a different name? Maybe "fromNominalHistogram" ?

Reply:
You mean to produce the fake nuisances for the ABCD fit with this using the "actionMap"? I think it's possible and I thought about it, but we have this other working implementation and it would be some additional work.